From 2ab185fbae69d3eb9d529ec29058cfd788ddc2f1 Mon Sep 17 00:00:00 2001
From: Pratyush Brahma <quic_pbrahma@quicinc.com>
Date: Mon, 23 Sep 2024 11:59:23 +0530
Subject: [PATCH 2/2] arm64: dts: qcom: sa8775p: Update memory map for SA8775P

New updates for memory map have brought in new carveouts
(viz. gunyah_md, tz-ffi and some pil carveouts) and
some size and base address updates to existing carveouts
(audio_mdf, trusted apps). Incorporate the updated memory map
in this change.

Signed-off-by: Pratyush Brahma <quic_pbrahma@quicinc.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/qcom/sa8775p.dtsi | 36 ++++++++++++++++++++++++---
 1 file changed, 33 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sa8775p.dtsi b/arch/arm64/boot/dts/qcom/sa8775p.dtsi
index 26f142dc06dc..68c6fc0d6479 100644
--- a/arch/arm64/boot/dts/qcom/sa8775p.dtsi
+++ b/arch/arm64/boot/dts/qcom/sa8775p.dtsi
@@ -426,6 +426,11 @@
 			no-map;
 		};
 
+		gunyah_md_mem: gunyah-md@91a80000 {
+			reg = <0x0 0x91a80000 0x0 0x80000>;
+			no-map;
+		};
+
 		aoss_backup_mem: aoss-backup@91b00000 {
 			reg = <0x0 0x91b00000 0x0 0x40000>;
 			no-map;
@@ -477,6 +482,16 @@
 			no-map;
 		};
 
+		q6_adsp_dtb_mem: q6-adsp-dtb@97a00000 {
+			reg = <0x0 0x97a00000 0x0 0x80000>;
+			no-map;
+		};
+
+		q6_gdsp0_dtb_mem: q6-gdsp0-dtb@97a80000 {
+			reg = <0x0 0x97a80000 0x0 0x80000>;
+			no-map;
+		};
+
 		pil_gdsp0_mem: pil-gdsp0@97b00000 {
 			reg = <0x0 0x97b00000 0x0 0x1e00000>;
 			no-map;
@@ -487,6 +502,16 @@
 			no-map;
 		};
 
+		q6_gdsp1_dtb_mem: q6-gdsp1-dtb@9b700000 {
+			reg = <0x0 0x9b700000 0x0 0x80000>;
+			no-map;
+		};
+
+		q6_cdsp0_dtb_mem: q6-cdsp0-dtb@9b780000 {
+			reg = <0x0 0x9b780000 0x0 0x80000>;
+			no-map;
+		};
+
 		pil_cdsp0_mem: pil-cdsp0@9b800000 {
 			reg = <0x0 0x9b800000 0x0 0x1e00000>;
 			no-map;
@@ -497,6 +522,11 @@
 			no-map;
 		};
 
+		q6_cdsp1_dtb_mem: q6-cdsp1-dtb@9d680000 {
+			reg = <0x0 0x9d680000 0x0 0x80000>;
+			no-map;
+		};
+
 		pil_cdsp1_mem: pil-cdsp1@9d700000 {
 			reg = <0x0 0x9d700000 0x0 0x1e00000>;
 			no-map;
@@ -512,8 +542,8 @@
 			no-map;
 		};
 
-		audio_mdf_mem: audio-mdf-region@ae000000 {
-			reg = <0x0 0xae000000 0x0 0x1000000>;
+		audio_mdf_mem: audio-mdf-region@ad000000 {
+			reg = <0x0 0xad000000 0x0 0x2000000>;
 			no-map;
 		};
 
@@ -568,7 +598,7 @@
 		};
 
 		trusted_apps_mem: trusted-apps@d1900000 {
-			reg = <0x0 0xd1900000 0x0 0x3800000>;
+			reg = <0x0 0xd1900000 0x0 0x1c00000>;
 			no-map;
 		};
 
-- 
2.17.1

